tests/unit/conftest.py

```diff
@@ -281,13 +281,225 @@
             },
         },
     },
+    "bad_combined_implementations": {
+        "step_1": {
+            "implementation": {
+                "name": "step_1_python_pandas",
+            },
+        },
+        "step_2": {
+            "implementation": {
+                "name": "step_2_python_pandas",
+            },
+        },
+        "step_3": {
+            "combined_implementation_key": "foo",
+        },
+        "step_4": {
+            "implementation": {
+                "name": "step_4_python_pandas",
+            },
+        },
+    },
 }
 
 INPUT_DATA_FORMAT_DICT = {
     "correct_cols": [["foo", "bar", "counter"], [1, 2, 3]],
     "wrong_cols": [["wrong", "column", "names"], [1, 2, 3]],
 }
 
+COMBINED_IMPLEMENTATION_CONFIGS = {
```

Comment:
I'm not sure if this is a good long term solution but maybe try making global variables like INPUT_DATA_FORMAT_DICT above but for multiple steps -- STEP_1_DICT, CONFIG_DICT, etc. Might make your life easier building up these large dicts in the future.

Reply:
Yeah, good thinking. I was thinking what I might want to do as part of https://jira.ihme.washington.edu/browse/MIC-5476 is to just actually break out all of these cases into yaml 

Another possibility is to just encode the *diff* with the default config and figure out the right way to update them

Reply:
+1 to the diff idea